File: app/backend/models/alerta_model.py
from sqlalchemy import Column, Integer, String, DateTime, SmallInteger, ForeignKey
from sqlalchemy.orm import relationship
from backend.database.connection import Base
from backend.models.estado_alerta_model import EstadoAlerta

class Alerta(Base):
    __tablename__ = 'Alerta'
    __table_args__ = {'implicit_returning': False}

    idAlerta = Column(Integer, primary_key=True, autoincrement=True)
    IdTransaccion = Column(Integer, nullable=True)
    IdTipoAlerta = Column(Integer, nullable=True)
    IdEstado = Column(Integer, ForeignKey('EstadoAlerta.IdEstado'), nullable=True)
    Asunto = Column(String(100), nullable=True)
    Mensaje = Column(String(5000), nullable=True)
    Medio = Column(String(50), nullable=True)
    IdPeriodicidad = Column(Integer, nullable=False)
    FechaInicio = Column(DateTime, nullable=True)
    FechaFin = Column(DateTime, nullable=True)
    Destinatarios = Column(String(5000), nullable=True)
    AudFecha = Column(DateTime, nullable=True)
    AudUsuario = Column(SmallInteger, nullable=True)
    Obs = Column(String(5000), nullable=True)
    DiasPers = Column(Integer, nullable=True)
    InfoExtra = Column(String(200), nullable=True)
    Url = Column(String(200), nullable=True)
    estado = relationship('EstadoAlerta', backref='alertas')

File: app/frontend/src/app/features/alertas/models/alerta.model.ts
export interface AlertaCreate {
  IdTransaccion?: number;
  IdTipoAlerta?: number;
  IdEstado?: number;
  Asunto?: string;
  Mensaje?: string;
  Medio?: string;
  IdPeriodicidad: number;  // Cambiado: ahora es obligatorio y es un ID numérico
  DiasPers?: number;       // Nuevo campo para días personalizados
  FechaInicio?: string;
  FechaFin?: string;
  Destinatarios?: string;  // Nuevo campo agregado
  Obs?: string;
  AudFecha?: string;
  AudUsuario?: number;
}

export interface Alerta extends AlertaCreate {
  idAlerta: number;
  DiasPers?: number;  // Asegurar que también esté en la interfaz Alerta
}

export interface AlertaUpdate {
  IdTransaccion?: number;
  IdTipoAlerta?: number;
  IdEstado?: number;
  Asunto?: string;
  Mensaje?: string;
  Medio?: string;
  IdPeriodicidad?: number;
  DiasPers?: number;       // Nuevo campo para días personalizados
  FechaInicio?: string;
  FechaFin?: string;
  Destinatarios?: string;
  Obs?: string;
  AudFecha?: string;
  AudUsuario?: number;
}
